File: base/freehep-connexion/src/main/java/org/freehep/connexion/rmi/ConnexionCallback.java
package org.freehep.connexion.rmi;

import java.rmi.Remote;
import java.rmi.RemoteException;

public interface ConnexionCallback extends Remote {

	public void addDevice(int device) throws RemoteException;

	public void removeDevice(int device) throws RemoteException;

	public void handleAxis(int device, int x, int y, int z, int rx, int ry,
			int rz) throws RemoteException;

	public void handleButtons(int device, int value, int buttons)
			throws RemoteException;
}

File: base/freehep-connexion/src/main/c++/ConnexionCallback.cpp

#include <iostream>

#include "ConnexionCallback.h"

using namespace std;

void ConnexionCallback::addDevice(int device) {
	cerr << "Connexion added device: " << device << endl;
}

void ConnexionCallback::removeDevice(int device) {
	cerr << "Connexion removed device: " << device << endl;
}
void ConnexionCallback::handleAxis(int device, int x, int y, int z, int rx, int ry, int rz) {
	cerr << "Connexion handle axis for device: " << device << " (" << x << ", " << y << ", " << z << ") (" << rx << ", " << ry << ", " << rz << ")" << endl;	
}
void ConnexionCallback::handleButtons(int device, int value, int buttons) {
	cerr << "Connexion handle buttons for device: " << device << " Value: " << value << " Buttons: " << buttons << endl;
}
